### ISSUE TITLE
Remove dead code related to stores under NULLCHK nodes in Z codegen

### ISSUE THREAD

**Issue by fjeremic** [eclipse/openj9#11318]
As pointed out on the OpenJ9 Slack (won't put a link here since it will expire) the following code seems dead:
https://github.com/eclipse/openj9/blob/f6fcc177cc1679c3fa7214c2cb1ea834ba318a24/runtime/compiler/z/codegen/J9TreeEvaluator.cpp#L5939-L5955

The reason being is that stores do not produce a value and hence cannot have a reference count > 1. This code can and should be removed.

**Comment by fjeremic** [comment]
This isn't completely true. There is a case where a store, for example a write barrier can appear underneath both a `TR::NULLCHK` and a `TR::compressedRefs` anchor:

```
n522n     NULLCHK on n510n [#32]                                                              [     0x3ff9d0b82e0] bci=[-1,344,-] rc=0 vc=0 vn=- li=- udi=- nc=1
n521n       awrtbari  org/apache/xerces/xni/QName.uri Ljava/lang/String;[#449  Shadow +20] [flags 0x607 0x0 ] (sharedMemory )  [     0x3ff9d0b8290] bci=[-1,344,-] rc=2 vc=0 vn=- li
n510n         ==>aloadi
n519n         ==>acalli
n510n         ==>aloadi
n524n     compressedRefs                                                                      [     0x3ff9d0b8380] bci=[-1,344,-] rc=0 vc=0 vn=- li=- udi=- nc=2
n521n       ==>awrtbari
n523n       lconst 0                                                                          [     0x3ff9d0b8330] bci=[-1,344,-] rc=1 vc=0 vn=- li=- udi=- nc=0
```

However I'm not convinced there is a need for a decrement and increment. I don't see anything bad that can happen from removing the code (famous last words). I'm going to launch a bunch of testing without it and see if anything breaks. If not I'm deleting the code.

**Comment by fjeremic** [comment]
Thanks for pointing that out. From my internal testing on Z there are no issues with removing the code. Going to make the same change on all the platforms and test again before proceeding to open the PR if no failures occur.

**Comment by 0xdaryl** [comment]
Just as a precaution, may I suggest that such a change not be merged until after the 0.24.0 release split on Dec 6?  It will give these changes lots of time to exercise in the builds before the 0.25.0 release.
